frontend/src/container/Sidebar.jsx

```diff
@@ -79,18 +80,31 @@ export class Sidebar extends React.Component {
         <WorkspaceListItem
           activeWorkspaceId={state.activeWorkspaceId}
           allowedAppList={space.sidebarEntryList}
+          foldChildren={!!state.foldedSpaceList.find(id => id === space.id)}
+          hasChildren={space.children.length > 0}
           label={space.label}
           level={spaceLevel}
           onClickAllContent={this.handleClickAllContent}
           userRoleIdInWorkspace={findUserRoleIdInWorkspace(props.user.userId, space.memberList, ROLE_LIST)}
           workspaceId={space.id}
           id={this.spaceItemId(space.id)}
+          onToggleFoldChildren={() => this.handleToggleFoldChildren(space.id)}
         />
-        {space.children.length !== 0 && this.displaySpace(spaceLevel + 1, space.children)}
+        {!state.foldedSpaceList.find(id => id === space.id) &&
+          space.children.length !== 0 &&
+          this.displaySpace(spaceLevel + 1, space.children)}
       </React.Fragment>
     )
   }
 
+  handleToggleFoldChildren = (id) => {
```

Comment:
Is the naming `id` right? I see that you compare this `id` to a variable called `space` in the function, seems fishy to me.

Reply:
I fixed `space`to `spaceId`